src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigAlignmentSignatureClassifier.java

```diff
@@ -0,0 +1,171 @@
+package org.broadinstitute.hellbender.tools.spark.sv.discovery.prototype;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.apache.logging.log4j.Logger;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignedContig;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignmentInterval;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.ChimericAlignment;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.RDDUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import scala.Tuple2;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+
+final class AssemblyContigAlignmentSignatureClassifier {
+
+    enum RawTypes {
+        InsDel,                 // 2 alignments, indicating ins/del, simple duplication expansion/contractions
+        IntraChrStrandSwitch,   // intra-chromosome strand switch breakpoints
+        TandemDupOrMEIBkpt,     // intra-chromosome tandem duplications without strand switch OR inter-chromosome MEI (with or without strand switch) breakpoints
```

Comment:
I'd refrain from calling things MEIs in the code until we know that's what they are. I can see why you might because most of the things that look like this will be MEIs in practice, but there will be other cases where it's not an MEI and it's confusing to call it an MEI before we've verified it (ie checked the insertion sequence against an MEI database or annotation set.